src/gui/vector/qgsstackeddiagramproperties.cpp

```diff
@@ -404,23 +411,15 @@ QMimeData *QgsStackedDiagramPropertiesModel::mimeData( const QModelIndexList &in
 
   QDataStream stream( &encodedData, QIODevice::WriteOnly );
 
-  for ( const QModelIndex &index : indexes )
+  // Create list of rows
+  const auto constIndexes = indexes;
+  for ( const QModelIndex &index : constIndexes )
   {
-    // each item consists of several columns - let's add it with just first one
     if ( !index.isValid() || index.column() != 0 )
       continue;
 
-    if ( QgsDiagramRenderer *diagram = mRenderers.at( index.row() ) )
-    {
-      QDomDocument doc;
-
-      QDomElement rootElem = doc.createElement( QStringLiteral( "diagram_mime" ) );
-      diagram->writeXml( rootElem, doc, QgsReadWriteContext() );
-      doc.appendChild( rootElem );
-      stream << doc.toString( -1 );
-    }
+    stream << index.row();
```

Comment:
This is not ideal, we're abusing mimeData to pass on a simple row index integer, instead of copying the actual diagram. This means we won't be able to do stuff like right-click -> copy diagram to then paste as new diagram, etc.

What's the rational behind this move?

Reply:
Addressed, I rebased to keep a clean commit history.

Now we have the fixes and still will be able to support the copying of diagrams.
Thanks for the suggestion.